main.py

```diff
@@ -347,65 +354,77 @@ def organize_result_directory(result_dir_path: str, rid: str) -> bool:
         )
         submission_time_dataset[0] = submission_time_str
+        # visualize option
         visualize_dataset = result_file.create_dataset("visualize", (1,), dtype="bool")
         visualize_dataset[0] = visualize
     # move the modified experiment file to the RID directory
     if visualize:
-        experiment_path = posixpath.join(result_dir_path, f"experiment_{submission_time_str}.py")
-        moved_experiment_path = posixpath.join(rid_dir_path, "modified_experiment.py")
-        shutil.move(experiment_path, moved_experiment_path)
+        src_experiment_path = posixpath.join(
+            result_dir_path, f"experiment_{submission_time_str}.py"
+        )
+        dst_experiment_path = posixpath.join(rid_dir_path, "modified_experiment.py")
+        shutil.move(src_experiment_path, dst_experiment_path)
     # remove the metadata file
     os.remove(metadata_path)
     return True
 
 
 @app.get("/result/")
 async def list_result_directory() -> List[int]:
-    """Post-processes the submitted experiments and returns the RID list.
+    """Post-processes the executed experiments and returns the RID list.
 
     It performs the following:
         1. Organize the result directory.
         2. Return the RID list.
 
     Returns:
-        A list with RIDs of the submitted experiments, sorted in an ascending order.
+        A list with RIDs of the executed experiments, sorted in an ascending order.
     """
-    rid_list = []
-    # read the most recently fetched RID
     result_dir_path = posixpath.join(configs["master_path"], configs["result_path"])
-    last_rid_path = posixpath.join(result_dir_path, "rid.json")
-    try:
-        with open(last_rid_path, encoding="utf-8") as last_rid_file:
-            last_rid = json.load(last_rid_file)
-    except FileNotFoundError:
-        last_rid = -1
-    # navigate to each RID directory
+    # navigate to each RID directory not organized yet
+    for item in os.listdir(result_dir_path):
+        item_path = posixpath.join(result_dir_path, item)
+        if posixpath.isdir(item_path) and item.startswith("_") and item[1:].isdigit():
+            rid = item[1:]
+            if organize_result_directory(result_dir_path, rid):
+                rid_dir_path = posixpath.join(result_dir_path, f"{rid}/")
+                os.rename(item_path, rid_dir_path)
+    # find all organized RID directories
+    rid_list = []
     for item in os.listdir(result_dir_path):
         item_path = posixpath.join(result_dir_path, item)
-        if posixpath.isdir(item_path) and item.isdigit():  # find a RID directory
-            rid = int(item)
-            if rid <= last_rid:  # alraedy organized
-                rid_list.append(rid)
-                continue
-            if organize_result_directory(result_dir_path, item):
-                rid_list.append(rid)
-    # update the most recently fetched RID
+        if posixpath.isdir(item_path) and item.isdigit():
+            rid_list.append(int(item))
```

Comment:
Isn't it better to do it within a single iteration?
I think this code is kind of duplicate.

Reply:
The returned values of `os.listdir(result_dir_path)` of each for loop may be different since a directory can be renamed in the first loop.
Thus, to merge them in one loop, there should be two cases:
1. The directory wasn't organized, but it is organized now.
2. The directory is already organized.

It may look more complicated, nevetheless is it okay?

Reply:
The only change is removing the heading "_", right?
Is the following code invalid?
```python
rid_list = []
for item in os.listdir(result_dir_path):
    item_path = posixpath.join(result_dir_path, item)
    if not posixpath.isdir(item_path):
        continue
    if item.startswith("_") and item[1:].isdigit():
        rid = item[1:]
        if organize_result_directory(result_dir_path, rid):
            rid_dir_path = posixpath.join(result_dir_path, f"{rid}/")
            os.rename(item_path, rid_dir_path)
            rid_list.append(int(rid))
    elif item.isdigit():
        rid_list.append(int(item))
rid_list.sort()
```

Reply:
As we discussed offline, please create an issue for using a subdirectory instead of the prefix "_" and ignore this review comment in this PR.